src/utils/actions/writeAction.ts

```diff
@@ -22,18 +27,41 @@ export const writeActionFromEvent = async (
   colonyAddress: string,
   actionFields: ActionFields,
 ): Promise<void> => {
-  const { transactionHash, blockNumber, timestamp } = event;
+  const { data } =
+    (await query<
+      GetColonyExtensionsByColonyAddressQuery,
+      GetColonyExtensionsByColonyAddressQueryVariables
+    >(GetColonyExtensionsByColonyAddressDocument, {
+      colonyAddress,
+    })) ?? {};
+  const extensionAddresses =
+    data?.listColonyExtensions?.items
+      .filter(notNull)
+      .map((colonyExtension) => colonyExtension.id) ?? [];
 
-  const actionType = actionFields.type ?? 'UNKNOWN';
-  const showInActionsList = await showActionInActionsList(
-    colonyAddress,
-    actionFields.initiatorAddress ?? '',
+  const isAddressInitiatorOrRecipient = (address: string): boolean =>
+    address === actionFields.initiatorAddress ||
+    address === actionFields.recipientAddress;
+  // @NOTE: If the action's initiator or recipient is an extension/network client, then we shouldn't show it in the action list.
+  const shouldShowAction = !(
+    extensionAddresses.find(isAddressInitiatorOrRecipient) ??
+    isAddressInitiatorOrRecipient(networkClient.address)
   );
 
-  verbose('Action', actionType, 'took place in Colony:', colonyAddress);
-  await mutate<CreateColonyActionMutation, CreateColonyActionMutationVariables>(
-    CreateColonyActionDocument,
-    {
+  if (shouldShowAction) {
```

Comment:
Our current approach for filtering the actions list is to set the `showInActionsList` flag, instead of conditionally adding the action to the db. 

It'd be good if we could follow that pattern here, i.e. move the logic inside the `showActionInActionsList` function.

Reply:
Are we getting something out of storing an action that we are never going to show? The way I see actions with `showInActionsList` set to `false` is that they _at some point_ could be displayed in the list.

Reply:
It is intentional, as far as I'm aware. We're using the db to keep track of everything, regardless of whether it gets shown to the user. For example, we don't remove motions that belong to an uninstalled voting rep extn. 

It might prove useful for things like recovering lost data. I'd check with Raul for full rationale.

Reply:
If that's the aim, makes sense to me